### PR TITLE
fix(ad): reject titles exceeding 65 characters

### PR DESCRIPTION
**Pull request by 1cu**
## ℹ️ Description

- Link to the related issue(s): Issue #979
- kleinanzeigen.de enforces a hard 65-character limit on ad titles. When a title exceeds this limit, the site silently rejects it, causing the bot to hang during publishing. This fix adds early validation so users get a clear, actionable error before any browser interaction occurs.

## 📋 Changes Summary

- Added `MAX_TITLE_LENGTH = 65` constant to `ad_model.py`
- Applied `Field(max_length=MAX_TITLE_LENGTH)` to the `title` field in `AdPartial`, leveraging Pydantic's built-in validation (semantic error type, auto-localized messages)
- Regenerated `schemas/ad.schema.json` — now includes `"maxLength": 65` on the title property
- Added parametrized test covering both boundary cases (65 chars accepted, 66 chars rejected)
- Cleaned up stale defensive `model_construct` tests that tested unreachable code paths

### ⚙️ Type of Change
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (adds new functionality without breaking existing usage)
- [ ] 💥 Breaking change (changes that might break existing user setups, scripts, or configurations)

## ✅ Checklist
- [x] I have reviewed my changes to ensure they meet the project's standards.
- [x] I have tested my changes and ensured that all tests pass (`pdm run test`).
- [x] I have formatted the code (`pdm run format`).
- [x] I have verified that linting passes (`pdm run lint`).
- [x] I have updated documentation where necessary.

## Summary by CodeRabbit

* **Bug Fixes**
  * Enforced both minimum (10) and maximum (65) title length limits and improved validation so titles outside bounds produce clear validation errors.

* **Tests**
  * Added unit tests covering title-length boundary behavior and updated related test setups.

### CODE REVIEW

File: src/kleinanzeigen_bot/model/ad_model.py [hunk 1]
```diff
@@ -155,7 +156,7 @@ def _validate_auto_price_reduction_constraints(price:int | None, auto_price_redu
 class AdPartial(ContextualModel):
     active:bool | None = _OPTIONAL()
     type:Literal["OFFER", "WANTED"] | None = _OPTIONAL()
-    title:str = Field(..., min_length = 10)
+    title:str = Field(..., min_length = 10, max_length = MAX_TITLE_LENGTH)
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/model/ad_model.py]: Thanks for the suggestion. This codebase intentionally uses `description=` only for internal/metadata fields whose purpose isn't self-evident from the field name (e.g., `repost_count`, `price_reduction_count`). Content fields like `title`, `description`, `category` are self-explanatory — their validation rules are expressed through Pydantic constraints (`min_length`, `max_length`) and surfaced to users via the error messages. Keeping this consistent with the existing convention.
